lms/djangoapps/learner_recommendations/views.py

```diff
@@ -124,6 +128,68 @@ def get(self, request, course_id):
         )
 
 
+class CrossProductRecommendationsView(APIView):
+    """
+    **Example Request**
+
+    GET api/learner_recommendations/cross_product/{course_id}/
+    """
+
+    def _empty_response(self):
+        return Response({"courses": []}, status=200)
+
+    def get(self, request, course_id):
+        """
+        Returns cross product recommendation courses
+        """
+        course_locator = CourseKey.from_string(course_id)
+        course_key = f'{course_locator.org}+{course_locator.course}'
+
+        associated_course_keys = get_cross_product_recommendations(course_key)
+
+        if not associated_course_keys:
+            return self._empty_response()
+
+        fields = [
+            "key",
+            "uuid",
+            "title",
+            "owners",
+            "image",
+            "url_slug",
+            "course_type",
+            "course_runs",
+            "location_restriction",
+        ]
+        course_data = [get_course_data(key, fields) for key in associated_course_keys]
+        filtered_courses = [course for course in course_data if course]
```

Comment:
- Adding the query string `marketable_course_runs_only': 1` will reduce the size of response and 
- return course runs for a course that are active. This will help recommend course runs that are active to users and not course runs that a user cannot enroll in. 

```suggestion
        course_data = [get_course_data(key, fields, querystring={'marketable_course_runs_only': 1}) for key in associated_course_keys]
        filtered_courses = [course for course in course_data if course and course_data.get("course_runs")]
```